src/Pos/Explorer/Web/Server.hs

```diff
@@ -103,43 +103,14 @@ getLastBlocks lim off = do
 
 getLastTxs :: ExplorerMode m => Word -> Word -> m [CTxEntry]
 getLastTxs (fromIntegral -> lim) (fromIntegral -> off) = do
-    let mkWhTx (txid, (tx, _, _)) = WithHash tx txid
-    localTxs <- fmap reverse $ topsortTxsOrFail mkWhTx =<< getLocalTxs
-    ts <- getCurrentTime
-
-    let neededLocalTxs = take lim $ drop off localTxs
-        localTxEntries = map (toTxEntry ts . view (_2 . _1)) neededLocalTxs
-        lenTxs = length localTxs
-        offLeft = off - lenTxs
-        nLeft = offLeft + lim
-
-    tip <- GS.getTip
-    let unfolder off lim h = do
-            when (lim <= 0) $
-                fail "Finished"
-            MaybeT (DB.getBlock h) >>= \case
-                Left gb -> unfolder off lim (gb ^. prevBlockL)
-                Right mb -> do
-                    let mTxs = mb ^. blockTxs
-                    if off >= length mTxs
-                        then return ([], (off - length mTxs, lim, mb ^. prevBlockL))
-                        else do
-                        txs <- topsortTxsOrFail identity $ map withHash $ toList mTxs
-                        blkSlotStart <- lift $ getSlotStart $
-                                        mb ^. gbHeader . gbhConsensus . mcdSlot
-                        let neededTxs = take lim $ drop off $ reverse txs
-                            blkTxEntries = map (toTxEntry blkSlotStart . whData) neededTxs
-                            offLeft = off - length mTxs
-                            nLeft = offLeft + lim
-                        return (blkTxEntries, (offLeft, nLeft, mb ^. prevBlockL))
-
-    blockTxEntries <- fmap concat $ flip unfoldrM (offLeft, nLeft, tip) $
-        \(o, l, h) -> runMaybeT $ unfolder o l h
-
-    return $ localTxEntries ++ blockTxEntries
+    txs <- allTxs
+    return $ take lim $ drop off $
```

Comment:
Changed. What about other occurrences of `return`? Do you prefer it replaced with `pure` everywhere?